rudder-web/src/main/scala/com/normation/rudder/web/snippet/HomePage.scala

```diff
@@ -332,6 +329,8 @@ class HomePage extends Loggable {
       n2               =  System.currentTimeMillis
       agentSoftEntries =  con.searchOne(acceptedNodesDit.SOFTWARE.dn, EQ(A_NAME, "rudder-agent"))
       agentSoftDn      =  agentSoftEntries.map(_.dn.toString).toSet
+      // Skip if there is no rudder-agent packages in software DN
+      if (! agentSoftDn.isEmpty)
```

Comment:
With that, you are returning Empty, which was forbiden in the Tablet Of Law for "wtf is the semantic of Empy in an Error handling, Failable sentence ?".
Please just return Full(Map()) in that case, which should be the case without the if, I think (agentSoftEntries == Seq(), agentSoftDn == Set(), agentSoft == Seq(), perhaps just add a if(agentSoftDn.isEmpty) Seq() else { ... } for the nodeEntries part).